azure/datalake/store/transfer.py

```diff
@@ -387,7 +387,8 @@ def _update(self, future):
                     merge_future = self._submit(
                         self._merge, self._adlfs, dst,
                         [chunk for chunk, _ in sorted(cstates.objects,
-                                                      key=lambda obj: obj[1])])
+                                                      key=lambda obj: obj[1])], 
+                        overwrite=self._parent._overwrite)
```

Comment:
I agree with both of you here, but I don't want to have to copy the value of the top level parameter (overwrite) everywhere. This value should be considered private in that once the parent is instantiated it should not be modified, as it is used to complete the upload (or download) operation.